docs/source/configuration/external_timelock_service_configs/timelock_server_config.rst

```diff
@@ -278,3 +278,29 @@ Note that because Timelock Server uses the OkHttp library, it is currently not c
    Thus, clients that are unable to use HTTP/2 may see a significant slowdown when the Timelock Server switches from an
    ``https`` connector to an ``h2`` connector. It may be possible to get around this by exposing multiple application
    connectors, though the AtlasDB team has not tested this approach.
+
+Non-Blocking Appender
+~~~~~~~~~~~~~~~~~~~~~
+
+We have experienced issues with Logback (the logging library which Dropwizard uses) under high load.
+This is because rolling of request logs required synchronization among threads that wanted to write logs; given high
+load it was possible for requests to build up and, eventually, servers being unable to respond to pings quickly enough.
+
+We thus implemented a ``NonBlockingFileAppenderFactory`` which never blocks when writing logs (even when files are
+rolled), though this could mean that some log lines may be dropped.
```

Comment:
I think it is a super small proportion:

• We only do this for request logs, not for service logs, and
• Logback has a number of threads that are allowed to queue up to do the logging. We only drop logs if the queue is exhausted, and that (based on what we've seen) only happens during a roll.

@nziebart could you confirm the above is correct?